src/app/components/modules/Settings.jsx

```diff
@@ -80,6 +82,55 @@ class Settings extends React.Component {
         this.setState({ nsfwPref, oldNsfwPref: nsfwPref });
     }
 
+    onDrop = (acceptedFiles, rejectedFiles) => {
+        if (!acceptedFiles.length) {
+            if (rejectedFiles.length) {
+                this.setState({
+                    progress: { error: 'Please insert only image files.' },
+                });
+                console.log('onDrop Rejected files: ', rejectedFiles);
+            }
+            return;
+        }
+        const file = acceptedFiles[0];
+        this.upload(file, file.name);
+    };
+
+    onOpenClick = imageName => {
+        this.setState({
+            imageInProgress: imageName,
+        });
+        this.dropzone.open();
+    };
+
+    upload = (file, name = '') => {
+        const { uploadImage } = this.props;
+        this.setState({
+            progress: { message: tt('settings_jsx.uploading_image') + '...' },
+        });
+        uploadImage(file, progress => {
+            if (progress.url) {
+                this.setState({ progress: {} });
+                const { url } = progress;
+                const image_md = `${url}`;
+                let field;
+                if (this.state.imageInProgress === 'profile_image') {
+                    field = this.state.profile_image;
+                } else if (this.state.imageInProgress === 'cover_image') {
+                    field = this.state.cover_image;
+                } else {
+                    return;
+                }
+                field.props.onChange(image_md);
+            } else {
+                this.setState({ progress });
+            }
+            setTimeout(() => {
+                this.setState({ progress: {} });
```

Comment:
@voorash progress is cleared unconditionally after 4s? Is there a success/failure signal we can use?

Reply:
Logic is the same as the upload image on the post page.  I will take a look at that tomorrow and get back to you.

Reply:
So here is the code from the reply editor. I didn't include the comment when copying it over. It pretty much just clears the message that says it is uploading as well.  
```
setTimeout(() => { this.setState({ progress: {} }) }, 4000) // clear message
```
I ran a test on this to make sure it did what I thought it does.  If it took over 4s for the server to respond the uploading message would be cleared.  If it ultimately failed it would show a failure message as the timeout does nothing but clear the message.  If on the other hand at 6 seconds it makes a connection and moves on with the upload the progress is automatically set from the user saga to show the percentage uploaded and all will proceed as normal.

During the actual upload the progress is tracked with this
```
xhr.upload.onprogress = function (event) {
        if (event.lengthComputable) {
            const percent = Math.round((event.loaded / event.total) * 100)
            progress({message: `Uploading ${percent}%`})
             console.log('Upload', percent)
        }
    }
```

Reply:
If the error message is not dismissable, the time-out might have some validity... If however our error display component is dismissable ... we might not need the timeout.